gr-blocks/python/blocks/qa_socket_pdu.py

```diff
@@ -100,6 +100,35 @@ def test_003 (self):
         #self.tb.connect(pdu_to_ts, head, sink)
         self.tb.run()
 
+    def test_004 (self):
+        # Test that the TCP server can stream PDUs <= the MTU size.
+        port = str(random.Random().randint(0, 30000) + 10000)
+        mtu = 10000
+        srcdata = tuple([x % 256 for x in xrange(mtu)])
+        data = pmt.init_u8vector(srcdata.__len__(), srcdata)
+        pdu_msg = pmt.cons(pmt.PMT_NIL, data)
+
+        self.pdu_source = blocks.message_strobe(pdu_msg, 500)
+        self.pdu_send = blocks.socket_pdu("TCP_SERVER", "localhost", port, mtu)
+        self.pdu_recv = blocks.socket_pdu("TCP_CLIENT", "localhost", port, mtu)
+        self.pdu_sink = blocks.message_debug()
+
+        self.tb.msg_connect(self.pdu_source, "strobe", self.pdu_send, "pdus")
+        self.tb.msg_connect(self.pdu_recv, "pdus", self.pdu_sink, "store")
+
+        self.tb.start()
+        while self.pdu_sink.num_messages() < 1:
+            time.sleep(0.1)
+        self.tb.stop()
```

Comment:
That's fair. I was trying to avoid another 1 second sleep but adding timeout code is even more unwieldy.